packages/taquito/src/contract/rpc-contract-provider.ts

```diff
@@ -677,6 +680,27 @@ export class RpcContractProvider extends Provider implements ContractProvider, S
       context
     );
   }
+
+  /**
+   *
+   * @description A failing_noop operation that is guaranteed to fail.
+   *
+   * @returns An operation handle with the result from the rpc node
+   *
+   * @param params failingNoop operation parameter
+   */
+  async failingNoop(params: FailingNoopParams) {
+    const publicKeyHash = await this.signer.publicKeyHash();
+
+    const prepared = await this.prepare.failingNoop({ ...params });
+    const content = prepared.opOb.contents.find(
+      (op) => op.kind === OpKind.FAILING_NOOP
+    ) as OperationContentsFailingNoop;
+    const opBytes = await this.forge(prepared);
+    const { hash, context, forgedBytes, opResponse } = await this.signAndInject(opBytes);
```

Comment:
Still don't think we actually want contract api to inject it (understand from integration test is worth injecting), thinking contract api return `{signContents: {...}, signature: ''}` will be good enough

Reply:
I kind of agree with An. Even if technically a failing_noop is designed to fail, from a user exp perspective, always having to capture an error just to grab a signature seems like bad user flow.